mParticle-Apple-SDK/Utils/NSDictionary+MPCaseInsensitive.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  NSDictionary+MPCaseInsensitive.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 1/29/25.
+//
+
+import Foundation
+public extension NSDictionary {
+
+    @objc func caseInsensitiveKey(_ key: String) -> String? {
+        var localKey = key
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                localKey = stringKey
+            }
+        }
+        return localKey
+    }
+
+    @objc func value(forCaseInsensitiveKey key: String) -> Any? {
+        var value : Any?
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                value = self[stringKey]
+            }
+        }
+        
+        return value
+    }
+
+    @objc func transformValuesToString() -> [String : Any] {
```

Comment:
This method really needs a dedicate unit test written in Objective-C that passes in a dictionary of all of the supported types from all the ifs and confirms the output strings. First make sure it passes against the original Obj-C implementation then against this one.